plugins/lookup/onepassword_ssh_key.py

```diff
@@ -0,0 +1,148 @@
+
+    def _get_raw(self, item_id, vault=None, token=None):
+        args = ["item", "get", item_id, "--format", "json"]
+        if vault is not None:
+            args = [*args, f"--vault={vault}"]
+
+        if self.service_account_token:
+            if vault is None:
+                raise AnsibleLookupError(
+                    "'vault' is required with 'service_account_token'"
+                )
+
+            environment_update = {
+                "OP_SERVICE_ACCOUNT_TOKEN": self.service_account_token
+            }
+            return self._run(args, environment_update=environment_update)
+
+        if token is not None:
+            args = [*args, to_bytes("--session=") + token]
+
+        return self._run(args)
+
+    def get_ssh_key(self, item_id, vault=None, token=None, ssh_format=False):
+        rc, out, err = self._get_raw(item_id, vault, token)
+
+        data = json.loads(out)
+
+        if data.get("category") != "SSH_KEY":
+            raise AnsibleLookupError(f"Item {item_id} is not SSH Key")
+
+        private_key_field = next(
+            (
+                field
+                for field in data.get("fields", {})
+                if field.get("id") == "private_key" and field.get("type") == "SSHKEY"
+            ),
+            None,
+        )
+        if not private_key_field:
+            raise AnsibleLookupError(f"No private key found for item {item_id}.")
+
+        if ssh_format:
+            return (
+                private_key_field.get("ssh_formats", {})
+                .get("openssh", {})
+                .get("value", "")
+            )
+        return private_key_field.get("value", "")
+
+
+class LookupModule(LookupBase):
+    def run(self, terms, variables=None, **kwargs):
+        self.set_options(var_options=variables, direct=kwargs)
+
+        ssh_format = kwargs.get("ssh_format")
+        vault = self.get_option("vault")
+        subdomain = self.get_option("subdomain")
+        domain = self.get_option("domain", "1password.com")
+        username = self.get_option("username")
+        secret_key = self.get_option("secret_key")
+        master_password = self.get_option("master_password")
+        service_account_token = self.get_option("service_account_token")
+        account_id = self.get_option("account_id")
+        connect_host = self.get_option("connect_host")
+        connect_token = self.get_option("connect_token")
+
+        op = OnePass(
+            subdomain=subdomain,
+            domain=domain,
+            username=username,
+            secret_key=secret_key,
+            master_password=master_password,
+            service_account_token=service_account_token,
+            account_id=account_id,
+            connect_host=connect_host,
+            connect_token=connect_token,
+            cli_class=OnePassCLIv2SSHKey,
+        )
+        op.assert_logged_in()
+
+        return [
+            op._cli.get_ssh_key(term, vault, ssh_format=ssh_format) for term in terms
```

Comment:
Is there a reason that you don't pass `token` here, while it is a valid parameter further up?

Reply:
The answer to your question is probably in the other onepassword plugin, where this one imports the code from ;-)

Reply:
The other plugins call `op.get_...`, not `op._cli.get_...`. The other function takes care of passing the token on.

So here the mechanism that passes in `token` doesn't work, while it does work fine in the other plugins.

Reply:
Correct.

Reply:
I think it would be better if the lookup simply uses `OnePassCLIv2` without changes, calls `op.get_raw()`, and then post-processes its result instead.

The onepassword_doc lookup also doesn't do this well, I gues it should also be fixed in a similar manner. For example, both onepassword_doc and this plugin totally ignore the changes from #7490 since they copy'n'pasted parts of onepassword.py.

Reply:
I created a PR for onepassword_doc: #9625.

For this plugin it's IMO better to simply use `OnePassCLIv2.get_raw` without modification and simply massage the result (as written above).

Reply:
That it much much better, thanks!!
I honestly saw the duplication, but simply followed the `onepassword_doc` since it's very similar to my plugin.

I'll use `self._add_parameters_and_run` instead of `self._run` once your PR get merged, is that okay?

Reply:
But why? You need the result of `get_raw()` and modify it, you don't need a different command, if I understand correctly? So why not call `get_raw()` and modify the result? That's the most straightforward approach.

Reply:
Looks much cleaner after [this](https://github.com/ansible-collections/community.general/pull/9580/commits/bd7c2407107224c1fcee8a5bc0e6173f6959c7a7) 🪶